lib/purl_fetcher/client/upload_files.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module PurlFetcher
+  class Client
+    # The file uploading part of a transfer
+    class UploadFiles
+      # @param [Hash<String,DirectUploadRequest>] file_metadata map of relative filepaths to file metadata
+      # @param [Hash<String,String>] filepath_map map of relative filepaths to absolute filepaths
+      def self.upload(file_metadata:, filepath_map:)
```

Comment:
Just put the absolute filepath in DirectUploadRequest to avoid the double hashes that are supposed to have identifical keys?